experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+        do so in experiment modification functions.
 
-        self.cpo_keys = [ "obj_id", "cpid", "txchannels", "rxchannels",  # REVIEW #26 Should this be named cpo_keys or something like radar_parameters, scan_parameters, scan_constructs. The idea being that an experiment will have a list of some lower level object (experiment_parameters will be a list of 'scans' or 'radar params' OR a scan parameter list will contain 1 or more sets of 'radar params')
-            "sequence", "pulse_shift", "mpinc", "pulse_len", "nrang", "frang", # REVIEW #1 Each of these should have a description - perhaps in a docstring
-            "intt", "intn", "beamdir", "scan", "scanboundf", "scanbound",
-            "txfreq", "rxfreq", "clrfrqf", "clrfrqrange", "xcf", "acfint",
-            "wavetype", "seqtimer" ]
+        exp_slice['beam_angle'] is going to be a list of possible beam directions for this
+        experiment slice in degrees off azimuth. It doesn't mean that these are the beam 
+        directions that will be used in the scan, those beams are in the beam_order list given by beamnumber. 
 
-        for num in range(self.cponum): # REVIEW #39 You might just want to instantiate the dict and set all to None above, avoiding the for loop. You can also use a dictionary comprehension: http://stackoverflow.com/questions/1747817/create-a-dictionary-with-list-comprehension-in-python - in this way you can increment the obj_id we think
-            cp_object = {} # I don't know what to call these anymore # REVIEW #6 should be labelled TODO
-            for key in self.cpo_keys:
-                cp_object[key] = None
-            cp_object["obj_id"] = num
-            cp_object["cpid"] = self.cpid
+    """  # TODO describe these
 
-            cpo_list.append(cp_object)
+    def __init__(self, cpid):
 
-        self.cpo_list=cpo_list
+        try:
+            assert isinstance(cpid, int)
+        except AssertionError:
+            errmsg = 'CPID must be a unique int'
+            raise ExperimentException(errmsg)
+
+        self.__cpid = cpid
+
+        self.__slice_dict = {}
+
+        self.__new_slice_id = 0
 
         # Load the config data
-        with open('../config.ini') as config_data: # REVIEW #15 An option would be to make the config file location string an argument, and supply this as the default. Also, should probably make a config handler class to error check and give a first pass at the config options. Similar to Keith's C++ code.
-            self.config=json.load(config_data)
+        self.__options = ExperimentOptions()
 
         # Next some metadata that you can change, with defaults.
-        # TODO: make default none and have a function that will
-        #   calculate something appropriate if left blank.
 
-        self.txctrfreq=12000 # in kHz.
-        self.txrate=12000000 # sampling rate, samples per sec
-        self.rxctrfreq=12000 # in kHz.
+        self.__txctrfreq = 12000  # in kHz.
```

Comment:
OK, I think txrate and rxrate should be only set in config. txctrfreq and rxctrfreq can be set by the user upon init (and not changeable, at least for now). So back to default values - I think the centre frequencies should have them, because most people will not have interest in changing it. If we set it at 13 MHz with 10 MHz txrate we give people access to 8 - 18 MHz. Can always throw an error if someone wants to send/receive 18-20 MHz.